### PR TITLE
scripts: pinctrl: update to MCUXpresso v13, and unify scripts

### PR DESCRIPTION
**Pull request by danieldegrasse**
Update scripts to MCUXpresso data source v13. Unify board and SOC
pin control data generation scripts, to simplify use by users interested
in generating board pin control files from MEX files.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

### CODE REVIEW

File: mcux/scripts/pinctrl/README.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+
+## Supported SOCs
+
+The following SOC lines are supported:
+- i.MX RT 10xx series
+- i.MX RT 11xx series
+- Kinetis K8x, K6x, K2x series
+- LPC 55sxx series
+
```
Comment (danieldegrasse) [inline mcux/scripts/pinctrl/README.md:30]: @DerekSnell  FYI- this is the list of SOCs I expect we should maintain support for with this script, please let me know if you think there are additional SOCs we should add (or any we should remove) from the official support list

File: mcux/scripts/pinctrl/README.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
+# MCUXpresso Pin Control Generation Scripts
+
+These scripts aid in the generation of pin control data NXP platforms. The
+scripts currently support most iMX.RT processors, and some LPC and Kinetis
```
Comment (DerekSnell) [inline mcux/scripts/pinctrl/README.md:4]:
```suggestion
scripts currently support most i.MX RT processors, and some LPC and Kinetis
```

File: mcux/scripts/pinctrl/README.md (hunk 1 above)
Comment (mmahadevan108) [reply mcux/scripts/pinctrl/README.md:30]: RT5xx & RT6xx?

Comment (danieldegrasse) [reply mcux/scripts/pinctrl/README.md:32]: > RT5xx & RT6xx?

Thanks, added

**Review by mmahadevan108 — APPROVED** [review]

**Review by decsny — APPROVED** [review]

**Review by DerekSnell — APPROVED** [review]

**Pull request merged**